Ghost_in_the_Shell/client_folder_react_app/ethisim/src/components/EditorComponents/FlowDiagram.js

```diff
@@ -0,0 +1,262 @@
+                border: '3px solid green',
+                borderRadius: '5%',
+                padding: 10,
+            },
+            position: componentData.position,
+        };
+    } else if (componentData.type === 'Generic') {
+        return {
+            id: componentData.id,
+            type: 'genericNode',
+            data: { label: componentData.title },
+            style: { border: '3px solid red', borderRadius: '5%', padding: 10 },
+            position: componentData.position,
+        };
+    } else if (componentData.type === 'Reflection') {
+        return {
+            id: componentData.id,
+            type: 'reflectionNode',
+            data: { label: componentData.title },
+            style: {
+                border: '3px solid purple',
+                borderRadius: '5%',
+                padding: 10,
+            },
+            position: componentData.position,
+        };
+    } else if (componentData.type === 'Conversation') {
+        return {
+            id: componentData.id,
+            type: 'conversationNode',
+            data: { label: componentData.title },
+            style: {
+                border: '3px solid blue',
+                borderRadius: '5%',
+                padding: 10,
+            },
+            position: componentData.position,
+        };
+    }
+}
+export default function FlowDiagram(props) {
+    const initialElements = mockComponentsFlowChart.components.map(
+        (componentData) => {
+            return initializeElements(componentData);
+        }
+    );
+
+    const [elements, setElements] = useState(initialElements);
+    const [isRemoveButtonDisabled, setIsRemoveButtonDisabled] = useState(true);
+    const [currentEdgeSelected, setCurrentEdgeSelected] = useState();
+
+    const classes = useStyles();
+
+    const nodeTypes = {
+        actionNode: actionNode,
+        reflectionNode: reflectionNode,
+        genericNode: genericNode,
+        conversationNode: conversationNode,
+    };
+
+    const graphStyles = { width: '100%', height: '500px', border: 'solid' };
+
+    const addEdge = (params, elements) => {
+        const { source, target } = params;
+        const id = 'edge-' + source + '-' + target;
+        //Node path to itself
+        if (source === target) {
+            return elements;
+        }
+        //Edge already exists
+        if (elements.find((elements) => elements.id === id)) {
+            return elements;
+        }
+        //Source already has an edge (A source node can't link to 2 different pages)
+        if (elements.find((elements) => elements.source === source)) {
+            return elements;
+        }
+        const newEdge = {
+            id,
+            source,
+            target,
+            animated: true,
+            arrowHeadType: 'arrowclosed',
+            style: { strokeWidth: '5px', arrowWidth: '2px' },
+        };
+        return elements.concat(newEdge);
+    };
+
+    const onConnect = (params) => {
```

Comment:
I wonder if it might make sense to put these in a functions directory, like functions/flowDiagramHelpers.js or something. Just to decrease the amount of code in the component itself. Just a thought, no idea if its actually better or not

Reply:
Would be more readable/easier to find, will implement